orchestration-shared/src/test/java/uk/gov/di/orchestration/shared/services/LogoutServiceTest.java

```diff
@@ -328,46 +325,24 @@ void shouldDeleteSessionFromRedisWhenNoCookieExists() {
         input.setRequestContext(contextWithSourceIp("123.123.123.123"));
 
         logoutService.handleLogout(
+                Optional.of(session),
                 Optional.empty(),
                 Optional.of(CLIENT_LOGOUT_URI),
                 Optional.of(STATE.getValue()),
                 auditUserWhenNoCookie,
-                Optional.empty(),
+                Optional.of(CLIENT_ID),
                 rpPairwiseId);
 
-        verify(sessionService, times(0)).deleteSessionFromRedis(SESSION_ID);
-        verifyNoInteractions(cloudwatchMetricsService);
+        verify(sessionService, times(1)).deleteSessionFromRedis(SESSION_ID);
+        verify(cloudwatchMetricsService).incrementLogout(Optional.of(CLIENT_ID));
         verify(auditService)
                 .submitAuditEvent(
                         LOG_OUT_SUCCESS,
-                        AuditService.UNKNOWN,
+                        CLIENT_ID,
                         auditUserWhenNoCookie,
                         AuditService.MetadataPair.pair("rpPairwiseId", rpPairwiseId.get()));
     }
 
-    @Test
-    void sessionsAreDeletedWhenDestroySessionsIsCalled() {
-
-        setupAdditionalClientSessions();
-
-        logoutService.destroySessions(session);
-
-        verify(backChannelLogoutService)
-                .sendLogoutMessage(
-                        argThat(withClientId("client-id-1")), eq(EMAIL), eq(INTERNAL_SECTOR_URI));
-        verify(backChannelLogoutService)
-                .sendLogoutMessage(
-                        argThat(withClientId("client-id-2")), eq(EMAIL), eq(INTERNAL_SECTOR_URI));
-        verify(backChannelLogoutService)
-                .sendLogoutMessage(
-                        argThat(withClientId("client-id-3")), eq(EMAIL), eq(INTERNAL_SECTOR_URI));
-
-        verify(clientSessionService).deleteStoredClientSession("client-session-id-1");
-        verify(clientSessionService).deleteStoredClientSession("client-session-id-2");
-        verify(clientSessionService).deleteStoredClientSession("client-session-id-3");
-        verify(sessionService, times(1)).deleteSessionFromRedis(SESSION_ID);
-    }
-
```

Comment:
related to the above, `destroySessions` was public because it was called in the case of a logout handler in the case of a logout request but this logic has been moved into the handle logout method. I should have made destroySessions private. Will fix.

Reply:
Done

Reply:
I still don't think we should be deleting this test - this is an important bit of logic that we want to make sure works. We should still be doing this testing somewhere

Reply:
I think most of this logic is covered in other tests but you're right, it does look like none of the other tests check that `backChannelLogoutService.sendLogoutMessage` is called or test for multiple sessions so I think it makes sure these are still covered

Reply:
Okay, I've added verifications for calls to `backChannelLogoutService.sendLogoutMessage` in the various other tests and a `sessionsAreAllDeletedOnLogout` tests which does much the same as the above but via a `handleLogout` call rather than a call to `logoutService.destroySessions`.